dynatrace/api/v2/settings/objects/permissions/settings_client.go

```diff
@@ -0,0 +1,182 @@
+
+type SettingsClient interface {
+	GetSchemaIDsWithOwnerBasedAccessControl(ctx context.Context) ([]string, error)
+	ListObjectsIDsOfSchema(ctx context.Context, schemaID string) ([]string, error)
+}
+
+type settingsClient struct {
+	client *rest.Client
+}
+
+// NewSettingsClient creates a new settings client for fetching schema IDs that support owner-based access control as well as listing object IDs of a given schema.
+func NewSettingsClient(client *rest.Client) *settingsClient {
+	return &settingsClient{client: client}
+}
+
+// GetSchemaIDsWithOwnerBasedAccessControl retrieves schema IDs that support owner-based access control.
+func (c *settingsClient) GetSchemaIDsWithOwnerBasedAccessControl(ctx context.Context) ([]string, error) {
+	response, err := c.getSchemas(ctx, "schemaId,ownerBasedAccessControl")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get schemas: %w", err)
+	}
+
+	var schemata schemaList
+	if json.Unmarshal(response.Data, &schemata) != nil {
+		return nil, fmt.Errorf("failed to unmarshal schema response: %w", err)
+	}
+
+	schemaIDs := []string{}
+	for _, schemaStub := range schemata.Items {
+		if schemaStub.OwnerBasedAccessControl {
+			schemaIDs = append(schemaIDs, schemaStub.SchemaID)
+		}
+	}
+	return schemaIDs, nil
+}
+
+func (c *settingsClient) getSchemas(ctx context.Context, fields string) (api.Response, error) {
+	response, err := c.client.GET(ctx, schemasEndpoint, rest.RequestOptions{QueryParams: url.Values{"fields": []string{fields}}})
+	if err != nil {
+		return api.Response{}, err
+	}
+
+	return api.NewResponseFromHTTPResponse(response)
+}
+
+// ListObjectsIDsOfSchema retrieves the IDs of objects for a given schema ID.
+// It tries first with admin access and then without, if the first request fails with a forbidden error.
+func (c *settingsClient) ListObjectsIDsOfSchema(ctx context.Context, schemaID string) ([]string, error) {
+	response, err := c.getObjectIDsFirstPageWithAdminRetry(ctx, schemaID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get first page of objects: %w", err)
+	}
+
+	var ol objectsList
+	if err := json.Unmarshal(response.Data, &ol); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal objects response: %w", err)
+	}
+
+	objectIDs := []string{}
+	for _, objectStub := range ol.Items {
+		objectIDs = append(objectIDs, objectStub.ObjectID)
+	}
+
+	for {
+		if ol.NextPageKey == "" {
+			break
+		}
+
+		response, err := c.getObjectsNextPage(ctx, ol.NextPageKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get next page of objects: %w", err)
+		}
+
+		ol = objectsList{}
+		if err := json.Unmarshal(response.Data, &ol); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal objects response: %w", err)
+		}
+
+		for _, objectStub := range ol.Items {
+			objectIDs = append(objectIDs, objectStub.ObjectID)
+		}
+	}
+
+	return objectIDs, nil
+}
+
+func (c *settingsClient) getObjectIDsFirstPageWithAdminRetry(ctx context.Context, schemaID string) (api.Response, error) {
```

Comment:
Depending on what gets merged first, either this one or the admin-access PR should align the behavior (re-use the DoWithAdminAccess function)

Reply:
Done, now it reuses `DoWithAdminAccess(...)`.